src/utils/markdown.rs

```diff
@@ -0,0 +1,243 @@
+//! Utils for working with the [MarkdownV2 message style.](https://core.telegram.org/bots/api#markdownv2-style)
+
+use std::string::String;
+
+/// Applies the bold font style to the string.
+/// Passed string will not be automatically escaped
+/// because it can contain nested markup.
+pub fn bold(s: &str) -> String {
+    wrap(s, "*", "*")
+}
+
+/// Applies the italic font style to the string.
+/// Can be safely used with `utils::markdown::underline()`.
+/// Passed string will not be automatically escaped
+/// because it can contain nested markup.
+pub fn italic(s: &str) -> String {
+    if s.starts_with("__") && s.ends_with("__") {
+        return wrap(&s[..s.len() - 1], "_", r"\r__");
+    }
+    wrap(s, "_", "_")
+}
+
+/// Applies the underline font style to the string.
+/// Can be safely used with `utils::markdown::italic()`.
+/// Passed string will not be automatically escaped
+/// because it can contain nested markup.
+pub fn underline(s: &str) -> String {
+    // In case of ambiguity between italic and underline entities
+    // ‘__’ is always greadily treated from left to right as beginning or end of underline entity,
+    // so instead of ___italic underline___ we should use ___italic underline_\r__,
+    // where \r is a character with code 13, which will be ignored.
+    if s.starts_with("_") && s.ends_with("_") {
+        return wrap(s, "__", r"\r__");
+    }
+    wrap(s, "__", "__")
+}
+
+/// Applies the strikethrough font style to the string.
+/// Passed string will not be automatically escaped
+/// because it can contain nested markup.
+pub fn strike(s: &str) -> String {
+    wrap(s, "~", "~")
+}
+
+/// Builds an inline link with an anchor.
+/// Escapes `)` and ``` characters inside the link url.
+pub fn link(url: &str, text: &str) -> String {
+    let mut out = String::with_capacity(url.len() + text.len() + 4);
+    out.push_str(wrap(text, "[", "]").as_str());
+    out.push_str(wrap(escape_link_url(url).as_str(), "(", ")").as_str());
+    out
+}
+
+/// Builds an inline user mention link with an anchor.
+pub fn user_mention(user_id: i32, text: &str) -> String {
+    link(format!("tg://user?id={}", user_id).as_str(), text)
+}
+
+/// Formats the code block. Escapes ``` and `\` characters inside the block.
+pub fn code_block(code: &str) -> String {
+    code_block_with_lang(code, "")
+}
+
+/// Formats the code block with a specific language syntax.
+/// Escapes ``` and `\` characters inside the block.
```

Comment:
```suggestion

/// Formats the code block with a specific language syntax.
///
/// Escapes ``` and `\` characters inside the block.
```